app/forms/hyrax/forms/dashboard/nest_collection_form.rb

```diff
@@ -83,12 +82,11 @@ def available_parent_collections
         # rerouting to new_dashboard_collection_path to add the new collection as
         # a child. Since we don't yet have a child collection, the valid? option can't be used here.
         def validate_add
-          if nestable?(parent)
-            nesting_within_maximum_depth
-          else
+          unless nestable?(parent)
```

Comment:
How are we guarding circular nesting? (e.g. A has member B has member C has member A)?

Reply:
According to the documentation, the graph query parser is "cyclic aware".

Reply:
on a current Hyrax, we allow circular nesting.

Reply:
I think it is being handled by the `parent_and_child_can_be_nested` validation on the form.

Reply:
@no-reply Can you give an example of currently allowed/working circular nesting?

Reply:
![circular-nesting](https://user-images.githubusercontent.com/1361776/189993147-eb8e9ad4-1605-40e3-84be-74e2bca4d567.png)

i was able to create this situation trivially in the UI.

Reply:
I wasn't able to reproduce this in nurax-dev.  Can you please post the exact steps that allowed this for you?  I'd like to reproduce this in nurax-dev then test the same steps with this PR to see how the behavior differs.